File: backendApp/src/main/java/com/cinemas/entities/Celebrity.java
package com.cinemas.entities;

import com.cinemas.enums.RoleCeleb;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Celebrity {
    @Id
    @GeneratedValue
    private Integer id;

    @Column
    private String name;

    @Column
    private LocalDate dateOfBirth;

    @Column
    private String slug;
    @Column
    private Integer view = 0;

    @Column(columnDefinition = "TEXT")
    private String biography;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column
    private RoleCeleb role;

    @Column
    private String image;

    @ManyToOne
    @JoinColumn(name = "country_id")
    private Country country;

    @ManyToMany(mappedBy = "actor")
    @JsonIgnore
    private List<Movie> moviesActor;

    @ManyToMany(mappedBy = "director")
    @JsonIgnore
    private List<Movie> moviesDirector;
}

File: fllmbooking_app/lib/data/models/celebrity.dart
enum RoleCeleb { ACTOR, DIRECTOR }

class Celebrity {
  int? id;
  String? name;
  String? description;
  DateTime? dateOfBirth;
  String? biography;
  RoleCeleb? role;
  String? image;
  int? countryId;
  String? slug;
  int? view;

  Celebrity({
    this.id,
    this.name,
    this.description,
    this.dateOfBirth,
    this.biography,
    this.role,
    this.image,
    this.countryId,
    this.slug,
    this.view,
  });

  factory Celebrity.fromJson(Map<String, dynamic> json) {
    return Celebrity(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      dateOfBirth: json['dateOfBirth'] != null
          ? DateTime.parse(json['dateOfBirth'])
          : null,
      biography: json['biography'],
      role: json['role'],
      image: json['image'],
      countryId: json['countryId'],
      slug: json['slug'],
      view: json['view'],
    );
  }

  static List<Celebrity> fromJsonList(List<dynamic> jsonList) {
    return jsonList.map((json) => Celebrity.fromJson(json)).toList();
  }
}
